go-controller/pkg/ovn/egressgw.go

```diff
@@ -144,7 +144,17 @@ func (oc *Controller) addGWRoutesForNamespace(namespace string, egress gatewayIn
 				nbctlArgs := []string{"--may-exist", "--policy=src-ip", "--ecmp-symmetric-reply",
 					"lr-route-add", gr, podIP.IP + mask, gw.String(), port}
 				if egress.bfdEnabled {
-					nbctlArgs = []string{"--may-exist", "--bfd", "--policy=src-ip", "--ecmp-symmetric-reply",
+					uuid, stderr, err := util.RunOVNNbctl(
+						"--format=csv", "--data=bare", "--no-heading", "--columns=_uuid", "find", "BFD", "logical_port="+port, "dst_ip=\""+gw.String()+"\"")
```

Comment:
I'm concerned about the performance impact of this. I think alternatively you could always just try to create the bfd. If one already exists you will get a transaction error like: 
```
[root@ovn-control-plane ~]# ovn-nbctl create BFD logical_port=rtoe-GR_ovn-worker dst_ip="172.18.0.4"
2021-09-28T15:53:28Z|00002|ovsdb_idl|WARN|transaction error: {"details":"Transaction causes multiple rows in \"BFD\" table to have identical values (rtoe-GR_ovn-worker and \"172.18.0.4\") for index on columns \"logical_port\" and \"dst_ip\".  First row, with UUID cbb1da43-6c99-4a53-8e38-2a58653830ff, existed in the database before this transaction and was not modified by the transaction.  Second row, with UUID 48f4fa3b-ae14-4268-8db9-94f3bc791bbb, was inserted by this transaction.","error":"constraint violation"}
ovn-nbctl: transaction error: {"details":"Transaction causes multiple rows in \"BFD\" table to have identical values (rtoe-GR_ovn-worker and \"172.18.0.4\") for index on columns \"logical_port\" and \"dst_ip\".  First row, with UUID cbb1da43-6c99-4a53-8e38-2a58653830ff, existed in the database before this transaction and was not modified by the transaction.  Second row, with UUID 48f4fa3b-ae14-4268-8db9-94f3bc791bbb, was inserted by this transaction.","error":"constraint violation"}
```

Or we could use a cache to store which gateways have bfds already (seems like a bad idea). Or we could wait for the OVN fix (seems like the best option?)

Reply:
> I'm concerned about the performance impact of this. I think alternatively you could always just try to create the bfd. If one already exists you will get a transaction error like:
> 
> ```
> [root@ovn-control-plane ~]# ovn-nbctl create BFD logical_port=rtoe-GR_ovn-worker dst_ip="172.18.0.4"
> 2021-09-28T15:53:28Z|00002|ovsdb_idl|WARN|transaction error: {"details":"Transaction causes multiple rows in \"BFD\" table to have identical values (rtoe-GR_ovn-worker and \"172.18.0.4\") for index on columns \"logical_port\" and \"dst_ip\".  First row, with UUID cbb1da43-6c99-4a53-8e38-2a58653830ff, existed in the database before this transaction and was not modified by the transaction.  Second row, with UUID 48f4fa3b-ae14-4268-8db9-94f3bc791bbb, was inserted by this transaction.","error":"constraint violation"}

yea I was trying to avoid this and inserted the find query, but you are absolutely right. Maybe doing 3 separate transactions is not the best way here.

> ovn-nbctl: transaction error: {"details":"Transaction causes multiple rows in \"BFD\" table to have identical values (rtoe-GR_ovn-worker and \"172.18.0.4\") for index on columns \"logical_port\" and \"dst_ip\".  First row, with UUID cbb1da43-6c99-4a53-8e38-2a58653830ff, existed in the database before this transaction and was not modified by the transaction.  Second row, with UUID 48f4fa3b-ae14-4268-8db9-94f3bc791bbb, was inserted by this transaction.","error":"constraint violation"}
> ```

I could catch this error like you are suggesting and ignore it if that happens., but the thing is we won't get a neat `uuid` output although I could parse it out of the error. I wish bfd create had a --may-exists, but the idea wasn't welcomed by ovn-team :D.

> 
> Or we could use a cache to store which gateways have bfds already (seems like a bad idea). 

yea seems like unnecessary complication

> Or we could wait for the OVN fix (seems like the best option?)

yes, we could do this as well and then we need to create different rpms all the way to 4.7? Adding the bfd session ourselves gives us more control in a way but I guess it doesn't matter as long as we understand what OVN is doing under the hood,

So we can either catch the transaction error like you are suggesting or wait for OVN fix.